cargo-dist/src/backend/installer/mod.rs

```diff
@@ -41,20 +41,15 @@ pub enum InstallerImpl {
 pub struct InstallerInfo {
     /// The path to generate the installer at
     pub dest_path: Utf8PathBuf,
-    /// App name to use (display only)
-    pub app_name: String,
-    /// App version to use (display only)
-    pub app_version: String,
     /// URL of the directory where artifacts can be fetched from
     pub base_url: String,
-    /// Artifacts this installer can fetch
-    pub artifacts: Vec<ExecutableZipFragment>,
     /// Description of the installer (a good heading)
     pub desc: String,
     /// Hint for how to run the installer
     pub hint: String,
-    /// Where to install binaries
-    pub install_path: JinjaInstallPathStrategy,
+    /// The release this is installing
+    #[serde(skip)]
+    pub release: ReleaseIdx,
 }
 
```

Comment:
This is the part that's over-aggressive and broke me. I hadn't appreciated that InstallerInfo just gets handed directly to each jinja template. We want each installer to lookup the Release and grab the PlatformSupport off of it. The reason this is late-bound like this is because we want PlatformSupport to be where installers grab things like shasums from, which in a sufficiently unified build would come after we compute the InstallerInfo (in principle we should be able to build binaries and then generate installers that fetch them in a single run of cargo-dist).

Reply:
Oh similarly "what glibc version do we need" and "linkage" are also late-bound like this. I think that's probably how I ended up doing a platform refactor when i was trying to do a linkage refactor -- in theory stuff like linkage is just like RuntimeConditions.

Reply:
I think if one had a mind to salvage this, they'd keep this change to include the ReleaseIdx but note delete all these other fields. Then write a shim that turns PlatformInfo into this "legacy" InstallerInfo/ExecutableZipFragment format. It won't include any of the "late-bound" info but we can then iteratively migrate each installer to using the latebound info (and also make sure it's populated).

With the goal of eventually dropping InstallerInfo/ExecutableZipFragment.

Reply:
i think another thing that overwhelmed me here was that i started convincing myself there should be huge overhauls to dist-manifest (there should) where maybe PlatformSupport replaces a ton of stuff in there (less obvious).